src/dlz_aws.cpp

```diff
@@ -196,10 +212,25 @@ isc_result_t dlz_lookup(
     }
   }
 
+  if (StringUtils::CaselessCompare(state->autoscaler_zone_name.c_str(), zone)) {
+    std::string clientAddr;
+    std::vector<std::string> nodes;
+    get_src_address(methods, clientinfo, &clientAddr);
+    if (state->client->TryResolveAutoscaler(name, clientAddr, &nodes)) {
+      size_t maxNodes = std::min(nodes.size(), state->num_asg_records);
+      for (const auto&& node : k_random<std::string>(nodes, maxNodes)) {
+        state->callbacks.putrr(lookup, "A", 120, node.c_str());
```

Comment:
We might want to make the timeout configurable (or hijack instance timeout), but we can save that for another CL.